kratos/processes/apply_ray_casting_interface_recognition_process.h

```diff
@@ -0,0 +1,197 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Daniel Diez
+//
+
+#pragma once
+
+// System includes
+#include <string>
+#include <iostream>
+
+// External includes
+
+// Project includes
+#include "processes/apply_ray_casting_process.h"
+
+namespace Kratos
+{
+///@addtogroup Kratos Core
+///@{
+
+///@name Kratos Classes
+///@{
+
+/// Applies ray casting to distinguish the color (like in/out) of each node in modelpart
+/** This class is used to define the which nodes are inside, outside or on the interface
+ *  of certain volume described by its contour
+ */
+template<std::size_t TDim>
+class KRATOS_API(KRATOS_CORE) ApplyRayCastingInterfaceRecognitionProcess : public ApplyRayCastingProcess<TDim>
+{
+
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of ApplyRayCastingInterfaceRecognitionProcess
+    KRATOS_CLASS_POINTER_DEFINITION(ApplyRayCastingInterfaceRecognitionProcess);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    using BaseType = ApplyRayCastingProcess<TDim>;
+
+    /**
+     * @brief Construct a new ApplyRayCastingProcess object using model
+     * and paramterts.
+     * Constructor without user defined extra rays epsilon, used to
+     * generate the extra rays when voting is required for coloring
+     * @param rVolumePart model part containing the volume elements
+     * @param rSkinPart model part containing the skin to compute
+     * the distance to as conditions
+     */
+    ApplyRayCastingInterfaceRecognitionProcess(
+        Model& rModel,
+        Parameters ThisParameters);
+
+    /**
+     * @brief Construct a new Apply Ray Casting Process object using an already created search strucutre
+     * @param TheFindIntersectedObjectsProcess reference to the already created search structure
+     * @param RelativeTolerance user-defined relative tolerance to be multiplied by the domain bounding box size
+     */
+    ApplyRayCastingInterfaceRecognitionProcess(
+        FindIntersectedGeometricalObjectsProcess& TheFindIntersectedObjectsProcess,
+        Parameters ThisParameters = Parameters());
+
+
+    /// Destructor.
+    ~ApplyRayCastingInterfaceRecognitionProcess() override 
+    {}
+
+    ///@}
+    ///@name Deleted
+    ///@{
+
+    /// Default constructor, needed for registry
+    ApplyRayCastingInterfaceRecognitionProcess() {}
```

Comment:
I'd move it from the Delete Doxygen section to the life cycle one.

Reply:
yep, makes sense